backend/infrahub/permissions/report.py

```diff
@@ -17,26 +18,51 @@
     from infrahub.permissions.types import KindPermissions
 
 
-def get_permission_report(
+def get_permission_report(  # noqa: PLR0911
     backend: PermissionBackend,
     permissions: AssignedPermissions,
     branch: Branch,
     node: MainSchemaTypes,
     action: str,
-    is_super_admin: bool = False,
-    can_edit_default_branch: bool = False,  # pylint: disable=unused-argument
+    global_permission_report: dict[GlobalPermissions, bool],
 ) -> BranchRelativePermissionDecision:
     is_default_branch = branch.name in (GLOBAL_BRANCH_NAME, registry.default_branch)
 
-    if is_super_admin:
+    if action != "view":
+        if node.kind in (InfrahubKind.ACCOUNTGROUP, InfrahubKind.ACCOUNTROLE, InfrahubKind.GENERICACCOUNT) or (
+            isinstance(node, NodeSchema) and InfrahubKind.GENERICACCOUNT in node.inherit_from
+        ):
+            return (
+                BranchRelativePermissionDecision.ALLOW
+                if global_permission_report[GlobalPermissions.MANAGE_ACCOUNTS]
+                else BranchRelativePermissionDecision.DENY
+            )
+        if node.kind in (InfrahubKind.BASEPERMISSION, InfrahubKind.GLOBALPERMISSION, InfrahubKind.OBJECTPERMISSION) or (
+            isinstance(node, NodeSchema) and InfrahubKind.BASEPERMISSION in node.inherit_from
+        ):
+            return (
+                BranchRelativePermissionDecision.ALLOW
+                if global_permission_report[GlobalPermissions.MANAGE_PERMISSIONS]
+                else BranchRelativePermissionDecision.DENY
+            )
+        if node.kind in (InfrahubKind.GENERICREPOSITORY, InfrahubKind.REPOSITORY, InfrahubKind.READONLYREPOSITORY) or (
+            isinstance(node, NodeSchema) and InfrahubKind.GENERICREPOSITORY in node.inherit_from
+        ):
+            return (
+                BranchRelativePermissionDecision.ALLOW
+                if global_permission_report[GlobalPermissions.MANAGE_REPOSITORIES]
+                else BranchRelativePermissionDecision.DENY
+            )
+
+    if global_permission_report[GlobalPermissions.SUPER_ADMIN]:
         return BranchRelativePermissionDecision.ALLOW
 
     decision = backend.report_object_permission(
         permissions=permissions["object_permissions"], namespace=node.namespace, name=node.name, action=action
     )
 
     # What do we do if edit default branch global permission is set?
-    # if can_edit_default_branch:
+    # if global_permission_report[GlobalPermissions.EDIT_DEFAULT_BRANCH]:
     #     decision |= PermissionDecisionFlag.ALLOW_DEFAULT
```

Comment:
can this comment be removed? should it be a Jira issue?

Reply:
You're right, logged IFC-934 about it.